app/views/components/_glance-metric.html.erb

```diff
@@ -1,16 +1,30 @@
 <%
+  measurement_display_labels = {Thousand: 'k', Million: 'm', Billion: 'b'}
   name ||= false
   figure ||= false
   period ||= false
-  measurement_display_label ||= ""
-  measurement_explicit_label ||= ""
   context ||= ""
   trend_percentage ||= ""
+
+  if figure
+    number = number_to_human(figure.gsub(/,/, '').to_i, :format => '%n', :precision => 3)
+    measurement_explicit_label = number_to_human(figure.gsub(/,/, '').to_i, :format => '%u', :precision => 3)
+    if figure.end_with? "%"
+      measurement_display_label = "%"
+    else
+      measurement_display_label = measurement_display_labels[measurement_explicit_label.to_sym]
+    end
+  end
```

Comment:
If possible, it would be good to move this code in to a presenter class, as that's more consistent with how the formatting for numbers is done currently. Maybe this one https://github.com/alphagov/content-data-admin/blob/master/app/presenters/single_content_item_presenter.rb , although I haven't fully looked at where the values are coming from, or what's happening to them.

Reply:
I think @cbaines has a good point that this should be considered 'complex' display logic, hence in a Rails world it should sit in a presenter so that it can potentially be shared between views. However, coming from a non-Rails world, I struggle to provide guidance on how it would ideally look like. I did it myself for the table component and tried to extract some bits in govuk_publishing_components and proved to be good practice.

Reply:
Cheers, both. I'm going to retire this PR and will raise another one with amendments later.